compiler/src/iree/compiler/DispatchCreation/FuseHorizontalContractions.cpp

```diff
@@ -47,62 +48,69 @@ struct FuseHorizontalContractionsPass final
-static bool checkOperationEquivalence(Operation *lhsOp, Operation *rhsOp) {
-  // During equivalence check, it would have been easier if `checkEquivalence`
-  // would just use `OpOperands *`. Since it takes `Value`s for now, just
-  // check that the values are the same as operands. This is potentially
-  // making the match too broad, but is an OK work-around for now.
-  // TODO(MaheshRavishankar): Fix upstream `checkEquivalence` signater in
-  // `OperationEquivalence::isEquivalentTo`.
-  llvm::SmallDenseSet<Value, 8> operands;
-  operands.insert(lhsOp->operand_begin(), lhsOp->operand_end());
-  operands.insert(rhsOp->operand_begin(), rhsOp->operand_end());
-
-  llvm::DenseMap<Value, Value> equivalentValues;
-  auto checkEquivalent = [&](Value lhsValue, Value rhsValue) {
-    if (operands.contains(lhsValue) && operands.contains(rhsValue)) {
-      return success();
-    }
-    return success(equivalentValues.lookup(lhsValue) == rhsValue ||
-                   equivalentValues.lookup(rhsValue) == lhsValue);
+static bool checkOperationEquivalence(Operation *aOp, Operation *bOp) {
+  auto aLinalgOp = dyn_cast<linalg::LinalgOp>(aOp);
+  auto bLinalgOp = dyn_cast<linalg::LinalgOp>(bOp);
+
+  if (!aLinalgOp || !bLinalgOp) {
+    return false;
+  }
+  // Contraction ops verifies that there are two operands and one result.
+  assert(linalg::isaContractionOpInterface(aLinalgOp) &&
+         linalg::isaContractionOpInterface(bLinalgOp) &&
+         "expected lhs and rhs to be contraction ops");
+
+  // CHeck that the LHS operand is the same. Can be relaxed in the future.
+  if (aLinalgOp.getDpsInputOperand(0)->get() !=
+      bLinalgOp.getDpsInputOperand(0)->get()) {
+    return false;
+  }
+
+  auto checkSameRankAndElementType = [](Value aVal, Value bVal) {
+    auto aType = dyn_cast<ShapedType>(aVal.getType());
+    auto bType = dyn_cast<ShapedType>(bVal.getType());
+    return aType && bType && aType.getRank() == bType.getRank() &&
+           aType.getElementType() == bType.getElementType();
   };
-  auto markEquivalent = [&](Value v1, Value v2) { equivalentValues[v1] = v2; };
-  return OperationEquivalence::isEquivalentTo(
-      lhsOp, rhsOp, checkEquivalent, markEquivalent,
-      /*flags=*/OperationEquivalence::IgnoreLocations);
-}
+  // Check that the RHS rank and element type are the same. We dont check the
+  // type cause we allow RHS to be transposes.
+  if (!checkSameRankAndElementType(aLinalgOp.getDpsInputOperand(1)->get(),
+                                   bLinalgOp.getDpsInputOperand(1)->get())) {
+    return false;
+  }
 
-/// Check that an operation is a `empty -> fill -> contraction`
-static bool isEmptyFillContractionDAGRootOp(
-    linalg::LinalgOp linalgOp,
-    std::optional<linalg::LinalgOp> seedContractionOp = std::nullopt) {
-  if (!linalg::isaContractionOpInterface(linalgOp)) {
+  // Check that the output rank and element type are the same. We dont check the
+  // type cause we allow RHS to be transposes.
```

Comment:
I don't follow this. How does transposed RHS impact the output type? Is it a copy-paste typo?

Reply:
yeah, copy-pasta.